### ISSUE TITLE
Incorrect data / IQN error in elastic-tube-1d in develop

### ISSUE THREAD

**Issue by MakisH** [precice/precice#1032]
**Describe your setup**

Operating system (e.g. Linux distribution and version): _Ubuntu 21.04_
preCICE Version: v2.2.1-211-gf1cb45f9
Tutorials 45eed (essentially v202104.1.1)

**Describe the problem**

Our elastic-tube-1d tutorial reports the following error when using preCICE from develop (not when using v2.2.1 as released):

```
(0) 14:45:12 [acceleration::BaseQNAcceleration]:202 in updateDifferenceMatrices: WARNING: The coupling residual equals almost zero. There is maybe something wrong in your adapter. Maybe you always write the same data or you call advance without providing new data first or you do not use available read data. Or you just converge much further than actually necessary.
(0) 14:45:12 [acceleration::BaseQNAcceleration]:231 in updateDifferenceMatrices: ERROR: Attempting to add a zero vector to the quasi-Newton V matrix. This means that the residual in two consecutive iterations is identical. There is probably something wrong in your adapter. Maybe you always write the same (or only incremented) data or you call advance without providing  new data first.
```

Full log: [solid-cpp.log](https://github.com/precice/precice/files/6642214/solid-cpp.log)

Most probably, this affects more cases.

**Step To Reproduce**
1. Run [the elastic-tube-1d tutorial](https://precice.org/tutorials-elastic-tube-1d.html) with `solid-cpp` and `fluid-cpp`.

**Expected behaviour**

Instead of exiting with an error, the `solid-cpp` solver should just start its time loop.

**Comment by KyleDavisSA** [comment]
I reproduce this problem. preCICE receives nothing from the solvers (not even a zero value). This is a piece of the output from the log file:

```
(0) 09:10:34 [acceleration::BaseQNAcceleration]:469 in concatenateCouplingData: [0m  Input value: 
(0) 09:10:34 [acceleration::BaseQNAcceleration]:469 in concatenateCouplingData: [0m  Input value: 
(0) 09:10:34 [acceleration::BaseQNAcceleration]:472 in concatenateCouplingData: [0m  Input data norm: 
```

Printing each value from the computation of `crossSectionLength` and `pressure` result in the correct values, so the issue is where the data is written to preCICE:

```
interface.writeBlockScalarData(crossSectionLengthID, chunkLength, vertexIDs.data(), crossSectionLength.data());
```
 Printing out the values of `crossSectionLength` in the StructureSolver results in the some values being displayed

```
for (auto const &i: crossSectionLength) {
	std::cout << i << " ";
}

1.12507 1.11269 1.10051 1.08854 1.07675 1.06516 1.05375 1.04253 1.03148 1.02061 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991 1.00991
```

Therefore, the problem could be with writing a block of data using `crossSectionLength.data()` ?

**Comment by KyleDavisSA** [comment]
For the Cpp solver:
Fails for both commit https://github.com/precice/precice/commit/f1cb45f92cd5148174461d6123c69b2f2c92d08b and commit https://github.com/precice/precice/commit/3ff0899092862ad6f081df033ca9b51d0041befa.
(Fails for both IQN-ILS and IQN-IMVJ)

For Python solver:
Fails for commit https://github.com/precice/precice/commit/f1cb45f92cd5148174461d6123c69b2f2c92d08b 
Passes for commit https://github.com/precice/precice/commit/3ff0899092862ad6f081df033ca9b51d0041befa. 

Can you confirm similar results? You mentioned 
> Instead of exiting with an error, the solid-cpp solver should just start its time loop.

however, it first runs 6 iterations before failing, so probably the QN implementation has been changed

**Comment by fsimonis** [comment]
Both cpp solvers precice/tutorials@5470dd0d99bd63d5b7d8564bcfc471c7e978c822 without modifications pass for 3ff0899 and fail for f1cb45f9.

So this seems to be related to the changes in #1004. As this was squashed into a single commit, I guess @BenjaminRodenberg needs to take this over. This is a good example of why huge commits are not a good idea.
Another question is why none of our tests caught this problem.

**Comment by BenjaminRodenberg** [comment]
> So this seems to be related to the changes in #1004. As this was squashed into a single commit, I guess @BenjaminRodenberg needs to take this over. This is a good example of why huge commits are not a good idea.

@KyleDavisSA has already found a candidate in #1036. I will wait for now, before actively working on this issue. See https://github.com/precice/precice/pull/1036#issuecomment-867532165.

> Another question is why none of our tests caught this problem.

Yes. This shows that we definitely need a test here!

**Comment by BenjaminRodenberg** [comment]
This issue should be solved via #1036 and a corresponding test has been added in #1050
